backend/tests/test_course_build_processor.py

```diff
@@ -0,0 +1,478 @@
+
+    mock_sqs_data = {"course_build_id": initial_course_build.id}
+
+    mock_settings = mocker.Mock()
+    setattr(mock_settings, "SQS_QUEUE", "testqueue")
+    setattr(mock_settings, "SQS_POLL_INTERVAL_MINS", "1")
+    mocker.patch(
+        "rope.api.processors.course_build_processor.settings",
+        mock_settings,
+    )
+
+    engine = create_engine("postgresql://pguser:pgpassword@localhost/ropedb")
+    mock_factory = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_db",
+        lambda: mock_factory,
+    )
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_moodle_user_role_by_shortname",
+        side_effects=["teacher", "student"],
+    )
+    mocker.patch(
+        "rope.api.routers.moodle.moodle_client.get_user_by_email",
+        return_value={
+            "id": 1,
+        },
+    )
+    mocker.patch(
+        "rope.api.processors.course_build_processor.create_course",
+        return_value={
+            "course_id": 77,
+            "course_enrolment_url": "https://enrolmenturl.com",
+            "course_enrolment_key": "amazing_enrolmentkey77",
+        },
+    )
+
+    sqs_stubber.add_response(
+        "get_queue_url",
+        {"QueueUrl": "https://testqueue"},
+        expected_params={"QueueName": "testqueue"},
+    )
+    sqs_stubber.add_response(
+        "receive_message",
+        {
+            "Messages": [
+                {
+                    "ReceiptHandle": "message1",
+                    "Body": json.dumps(mock_sqs_data),
+                }
+            ]
+        },
+        expected_params={
+            "QueueUrl": "https://testqueue",
+            "MaxNumberOfMessages": 10,
+            "WaitTimeSeconds": 20,
+        },
+    )
+    sqs_stubber.add_response(
+        "delete_message",
+        {},
+        expected_params={
+            "QueueUrl": "https://testqueue",
+            "ReceiptHandle": "message1",
+        },
+    )
+
+    sqs_stubber.activate()
+    mocker_map = {"sqs": sqs_client}
+    mocker.patch("boto3.client", lambda client: mocker_map[client])
+    mocker.patch("sys.argv", [""])
+    course_build_processor.main()
+
+    get_sessionmaker = course_build_processor.get_db()
+    with get_sessionmaker() as session:
+        updated_course_build = (
+            session.query(CourseBuild)
+            .filter(CourseBuild.id == initial_course_build.id)
+            .all()
+        )
+
+        assert len(updated_course_build) == 1
+        assert updated_course_build[0].status == "completed"
+        assert updated_course_build[0].course_id == 77
+        assert updated_course_build[0].course_enrollment_key == "amazing_enrolmentkey77"
+        assert (
+            updated_course_build[0].course_enrollment_url == "https://enrolmenturl.com"
+        )
+
+    sqs_stubber.assert_no_pending_responses()
+
+
+def test_non_existing_course_build(mocker):
+    with pytest.raises(
```

Comment:
I think some of my comments in the code are going to impact this test case. In general, we want to avoid paths that cause exceptions to escape the code since that means the process will cycle. So, if we want to validate error paths we need to assert on something else if possible. I think here all we really care about functionally is that the message is not deleted. We can use `assert_no_pending_responses()` from the stubber for that.